iree/compiler/Conversion/HLOToLinalg/test/linalg_tensor_to_buffer.mlir

```diff
@@ -163,3 +163,125 @@ module {
+
+module {
+  func @store_reshape_src_and_result_0() {
+    %c0 = constant 0 : index
+    %0 = hal.interface.load.tensor @legacy_io::@arg0,
+                                   offset = %c0 : tensor<2x4xf32>
+    %1 = linalg.generic
+           {args_in = 1 : i64, args_out = 1 : i64,
+            indexing_maps = [#map0, #map0],
+            iterator_types = ["parallel", "parallel"]} %0 {
+    ^bb0(%arg0: f32):  // no predecessors
+      %2 = tanh %arg0 : f32
+      linalg.yield %2 : f32
+    }: tensor<2x4xf32> -> tensor<2x4xf32>
+    %3 = linalg.tensor_reshape %1 [#map1, #map2] : tensor<2x4xf32> into tensor<1x2x4xf32>
+    hal.interface.store.tensor %3, @legacy_io::@ret1, offset = %c0 : tensor<1x2x4xf32>
+    hal.interface.store.tensor %1, @legacy_io::@ret0, offset = %c0 : tensor<2x4xf32>
+    return
+  }
+  hal.interface @legacy_io attributes {sym_visibility = "private"} {
+    hal.interface.binding @arg0, set=0, binding=0,
+                                 type="StorageBuffer", access="Read"
+    hal.interface.binding @ret0, set=0, binding=7,
+                                 type="StorageBuffer", access="Write|Discard"
+    hal.interface.binding @ret1, set=0, binding=8,
+                                 type="StorageBuffer", access="Write|Discard"
+  }
+}
+
+// CHECK-LABEL: func @store_reshape_src_and_result_0
+//       CHECK: %[[T0:.*]] = iree.placeholder {{.*}} @legacy_io::@ret1
+//       CHECK: %[[T1:.*]] = linalg.reshape %[[T0]]
+//       CHECK: %[[T2:.*]] = iree.placeholder {{.*}} @legacy_io::@ret0
+//       CHECK: %[[T3:.*]] = iree.placeholder {{.*}} @legacy_io::@arg0
+//       CHECK: linalg.generic {{.*}} %[[T3]], %[[T1]]
+//       CHECK: linalg.copy(%[[T1]], %[[T2]])
+
+// -----
+
+#map0 = affine_map<(d0, d1) -> (d0, d1)>
+#map1 = affine_map<(d0, d1, d2) -> (d0, d1)>
+#map2 = affine_map<(d0, d1, d2) -> (d2)>
+
+module {
+  func @store_reshape_src_and_result_1() {
+    %c0 = constant 0 : index
+    %0 = hal.interface.load.tensor @legacy_io::@arg0,
+                                   offset = %c0 : tensor<2x4xf32>
+    %1 = linalg.generic
+           {args_in = 1 : i64, args_out = 1 : i64,
+            indexing_maps = [#map0, #map0],
+            iterator_types = ["parallel", "parallel"]} %0 {
+    ^bb0(%arg0: f32):  // no predecessors
+      %2 = tanh %arg0 : f32
+      linalg.yield %2 : f32
+    }: tensor<2x4xf32> -> tensor<2x4xf32>
+    %3 = linalg.tensor_reshape %1 [#map1, #map2] : tensor<2x4xf32> into tensor<1x2x4xf32>
+    hal.interface.store.tensor %1, @legacy_io::@ret0, offset = %c0 : tensor<2x4xf32>
+    hal.interface.store.tensor %3, @legacy_io::@ret1, offset = %c0 : tensor<1x2x4xf32>
+    return
+  }
+  hal.interface @legacy_io attributes {sym_visibility = "private"} {
+    hal.interface.binding @arg0, set=0, binding=0,
+                                 type="StorageBuffer", access="Read"
+    hal.interface.binding @ret0, set=0, binding=7,
+                                 type="StorageBuffer", access="Write|Discard"
```

Comment:
nit: make the binding values contiguous, eg, `s/binding=7/binding=1`

plz also update other test cases.

This was copied from other snippet and I forgot to update...